From c69c3105073a534015ca3d9edf9b6902836dc02d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 26 Feb 2016 04:39:33 +0100
Subject: [PATCH] Fix checking available features

Indicators and advanced menu are need to be checked in the update_panel_layout_features function, othervise it's always enabled if the layouts files are availabe.
---
 mate-tweak | 19 ++++++++-----------
 1 file changed, 8 insertions(+), 11 deletions(-)

diff --git a/mate-tweak b/mate-tweak
index a31622b..7d8f058 100755
--- a/mate-tweak
+++ b/mate-tweak
@@ -409,8 +409,14 @@ class MateTweak:
         if self.panel_layout_exists(panel_layout + '-fresh') and \
            self.panel_layout_exists(panel_layout + '-indicators') and \
            self.panel_layout_exists(panel_layout + '-indicators-fresh'):
-            self.builder.get_object('checkbutton_indicators').props.sensitive = True
-            self.builder.get_object('checkbutton_advanced_menu').props.sensitive = True
+            if not self.indicators_available:
+                self.builder.get_object('checkbutton_indicators').props.sensitive = False
+            else:
+                self.builder.get_object('checkbutton_indicators').props.sensitive = True
+            if not self.mate_menu_available:
+                self.builder.get_object('checkbutton_advanced_menu').props.sensitive = False
+            else:
+                self.builder.get_object('checkbutton_advanced_menu').props.sensitive = True
         else:
             self.builder.get_object('checkbutton_indicators').props.sensitive = False
             self.builder.get_object('checkbutton_advanced_menu').props.sensitive = False
@@ -664,15 +670,6 @@ class MateTweak:
 
 
     def make_list_of_panel_layouts(self):
-        if not self.indicators_available:
-            self.builder.get_object('checkbutton_indicators').props.sensitive = False
-        else:
-            self.builder.get_object('checkbutton_indicators').props.sensitive = True
-        if not self.mate_menu_available:
-            self.builder.get_object('checkbutton_advanced_menu').props.sensitive = False
-        else:
-            self.builder.get_object('checkbutton_advanced_menu').props.sensitive = True
-
         # Panel layouts
         panels = Gtk.ListStore(str, str)
 
-- 
2.7.1

